src/compiler/checker.ts

```diff
@@ -14965,7 +14965,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             result = appendIfUnique(result, getDeclaredTypeOfTypeParameter(node.symbol));
         }
         return result?.length ? result
-            : isFunctionDeclaration(declaration) ? getSignatureOfTypeTag(declaration)?.typeParameters
+            : isFunctionDeclaration(declaration) ? getSignatureOfTypeTag(declaration, /*allowMembers*/ false)?.typeParameters

```

Comment:
```ts
/** @type {{(arg1: string): void;}} */
function Named() {
	this.x = 1;
};
Named.foo = 10;
```

This is currently treated as a JS class and the extra expando property is allowed too. The generated DTS file looks like this:
```ts
declare function Named(arg1: string): void;
declare class Named {
    x: number;
}
declare namespace Named {
    let foo: number;
}
```

So... it would potentially also make sense to allow members here.

Reply:
This is already using `allowMember: true` with the new code - and we need that because otherwise the signature would not be retrieved at all (and thus its type parameters wouldn't be retrieved either) when `declaration` has members